Src/BootCamp.Chapter/Program.cs

```diff
@@ -1,11 +1,26 @@
 ﻿using System;
+using BootCamp.Chapter.Hints;
+using System.Collections.Generic;
 
 namespace BootCamp.Chapter
 {
     class Program
     {
         static void Main(string[] args)
         {
+            Teacher<Math> like = new Teacher<Math>();
+            like.ProduceMaterial(); //welcome to Math lesson
+
+            var yanan = new Student(123);
+            yanan.LearnFrom<Teacher<Math>,Math>(like); //student 123 learn from Math teacher
```

Comment:
Seems to be a bit redundant. I'd say it would be enough to just pass a teacher- after all the teacher knows only one subject- so it can provide it to the student itself. This would work though if teacher were teaching multiple subjects- in this case they are not.